source/javascripts/services/_stepSourceService.js.erb

```diff
@@ -11,6 +11,35 @@ angular.module("BitriseWorkflowEditor").factory("stepSourceService", function($q
 		gitSteps: []
 	};
 
+	var getLibraryForStep = function(step) {
+		if (!step.isLibraryStep()) {
+			return null;
+		}
+
+		var library = _.find(stepSourceService.libraries, {
+			url: step.libraryURL
+		});
+
+		if (!library) {
+			throw new Error("Library can not found for step");
+		}
+
+		return library;
+	}
+
+	var configureStepFromLib = function(step, library) {
+		var requestedVersion = step.version ? step.version : library.latestStepVersions[step.id];
+		var shouldLoadOnlyStepIcon = !library.steps[step.id][requestedVersion];
+
+		if (shouldLoadOnlyStepIcon) {
+			requestedVersion = library.latestStepVersions[step.id];
+			step.assetUrls(library.steps[step.id][requestedVersion].defaultStepConfig.asset_urls);
```

Comment:
Couldn't this fail is `library.steps[step.id][requestedVersion]` is `undefined`?

Reply:
well `requestedVersion` is coming from the library that means we don't have trust in library but that's in our domain. IMHO that would be an over pre-emptive coding. WDYT?